fix(txt-suffix): avoid crash on domain without dot

titilambert opened the pull request:
<!--
    Please read https://github.com/kubernetes-sigs/external-dns#contributing before submitting
    your pull request. Please fill in each section below to help us better prioritize your pull request. Thanks!
-->

**Description**

<!-- Please provide a summary of the change here. -->
This PR fix issue when suffix is used and the domain has no dot

<!-- Please link to all GitHub issue that this pull request implements(i.e. Fixes #123) -->
Fixes #1668

**Checklist**

- [X] Unit tests updated
- [N/A] End user documentation updated

mloiseleur commented on registry/txt.go:
```diff
@@ -406,6 +406,10 @@ func (pr affixNameMapper) toEndpointName(txtDNSName string) (endpointName string
 		domainWithSuffix := strings.Join(DNSName[:1+dc], ".")
 
 		r, rType := pr.dropAffixExtractType(domainWithSuffix)
+		if 1+dc >= len(DNSName) {
+			// If it's a domain without dot
```
```suggestion
		if strings.Contains(DNSName, ".") {
```

mloiseleur commented:
/retitle fix(txt-suffix): avoid crash on domain without dot

mloiseleur commented:
/ok-to-test
@titilambert Wdyt of adding a log, too ?

titilambert commented:
> /ok-to-test @titilambert Wdyt of adding a log, too ?

Good idea ! I will add a commit later today.

titilambert commented:
> /ok-to-test @titilambert Wdyt of adding a log, too ?

Done !

titilambert commented:
> > [Keywords](https://help.github.com/articles/closing-issues-using-keywords) which can automatically close issues and at(@) or hashtag(#) mentions are not allowed in commit messages.
> > **The list of commits with invalid commit messages**:
> > ```
> > * [85a4b15](https://github.com/kubernetes-sigs/external-dns/commits/85a4b154f160819dcc8f469414d5e90fde1d64f8) Fix [wrong host entry crashes the whole external-dns pod #1668](https://github.com/kubernetes-sigs/external-dns/issues/1668)
> > ```
> 
> @titilambert You have to fix this commit, too.

Sorry, I missed it. I just squashed, changed commit message and force pushed.

titilambert commented:
> > > [Keywords](https://help.github.com/articles/closing-issues-using-keywords) which can automatically close issues and at(@) or hashtag(#) mentions are not allowed in commit messages.
> > > **The list of commits with invalid commit messages**:
> > > ```
> > > * [85a4b15](https://github.com/kubernetes-sigs/external-dns/commits/85a4b154f160819dcc8f469414d5e90fde1d64f8) Fix [wrong host entry crashes the whole external-dns pod #1668](https://github.com/kubernetes-sigs/external-dns/issues/1668)
> > > ```
> > 
> > 
> > @titilambert You have to fix this commit, too.
> 
> Sorry, I missed it. I just squashed, changed commit message and force pushed.

Done

mloiseleur commented:
/lgtm
/assign @Raffo 

@Raffo: I'll let you decide if we should keep or not those debug logs.

titilambert commented:
> I don't think they are super informative and won't really help in a running system. I understand the use for development, but I think we should remove them. Can you take care of this @titilambert ?

done !

titilambert commented:
@Raffo @mloiseleur I think I completed everything, so It should be ready for review.

Thanks again !

Raffo commented:
/approve
/lgtm

Raffo commented:
/hold

Raffo commented:
Removing hold once I've seen CI pass, I didn't realize that it didn't re-run.
